note.h

```diff
@@ -55,7 +55,11 @@
 #define ERRDBG
 #endif
 
+#ifdef NOTE_C_LOW_MEM
```

Comment:
How much value is there in making this distinction? Do we know how much memory we are saving by switching down to `float` from `double`?

This feels very similar to the `JINTEGER` discussion. My gut instinct is to only support `double`, unless we have a compelling reason to do otherwise.

Reply:
So the thinking is that most small MCUs define double and float to the same thing: a single-precision (32-bit) float. That's what drives the logic on lines 28-40. The primary motivator here is testing. If I'm running the unit tests on my 64-bit Linux machine, I can define NOTE_C_LOW_MEM and run the tests using 32-bit floats instead of doubles, without the need for an actual platform like an Uno where there's ONLY 32-bit floats (no doubles).

Reply:
In that case, it seems as if it should be under a test specific define, instead of the `NOTE_C_LOW_MEM` define.

Reply:
Yeah I got there eventually, too, after giving it more thought. I'll tweak this PR accordingly.

Reply:
FWIW I would caution against assuming that UNO's broken compiler is representative of "most" MCU's.  I don't know who "most" really is.  

For example, I could argue that whatever compilers are used for native PIC or MSP430 probably define the majority.   

Net-net, having specific defines would be the best, with default internal behaviors being the "most correct, least gotchas" of 64-bit int64_t and 64-bit double.